src/raft.rs

```diff
@@ -525,8 +530,7 @@ impl<T: Storage> Raft<T> {
         let ents = self.raft_log.entries(pr.next_idx, self.max_msg_size);
         let mut m = Message::new();
         m.set_to(to);
-        if term.is_err() || ents.is_err() {
-            // send snapshot if we failed to get term or entries
+        if term.is_err() || ents.is_err() || pr.pending_request_snapshot != INVALID_INDEX {
```

Comment:
Better move `pending_request_snapshot` check forward to avoid unnecessary loading entries.